brave/browser-laptop
# Updated General Preferences

Updated General Preferences

- Show home button now disables itself when home page is cleared (null/empty)
- Removed appearance settings; moved to under "My home page is" per the mockup on https://github.com/brave/browser-laptop/issues/2108
- Bookmarks toolbar switches were changed into a single drop down
- Small refactor done to settings.js (with regards to default settings)

Unit tests are in place and working + I've manually at least two of the scenarios outlined in the attached steps

Fixes https://github.com/brave/browser-laptop/issues/2108
Fixes https://github.com/brave/browser-laptop/issues/3008
Fixes https://github.com/brave/browser-laptop/issues/2111
Possibly fixes https://github.com/brave/browser-laptop/issues/3544 (auditor discretion)

## Auditors
@bbondy, @ayumi, @cezaraugusto - small changes EXCEPT the new config value for BTB. This only had two touchpoints (main.js, contextMenus.js) versus the password manager changes

@BrendanEich - you had previously reviewed the settings code; LMK how the changes in `./js/settings.js` look

## Upgrade test plan
1. Install a previous version of Brave (0.12.4 or older); ENABLE favicons and ENABLE show only favicons
2. Upgrade to this version (0.12.5?). Confirm it consolidates preference to "Favicons only" in preferences drop down
3. Install a previous version of Brave (0.12.4 or older); ENABLE favicons but DISABLE show only favicons
4. Upgrade to this version (0.12.5?). Confirm it consolidates preference to "Text and Favicons" in preferences drop down
5. Install a previous version of Brave (0.12.4 or older); DISABLE favicons and DISABLE show only favicons
6. Upgrade to this version (0.12.5?). Confirm it consolidates preference to "Text only" in preferences drop down
7. Uninstall Brave completely; move session file out of the way

## Regular test plan
### consolidation of bookmarks toolbar settings
1. Install a clean version of Brave (0.12.5?)
2. Launch preferences, confirm preference defaults to "Text only"
3. Confirm no favicons show (only text):
  - at the root level of the  bookmarks toolbar
  - when expanding a folder inside the bookmarks toolbar
4. Repeat step 2/3 for the other two settings. NOTE: show only favicon does not apply to items inside bookmark folders

### bookmark toolbar enable/disable change
5. Go to preferences and toggle "Always show the bookmarks toolbar"
6. If disabled, notice how drop down is now disabled too (not allowing you to change)
7. When enabled, notice how drop down is re-enabled

### disabling of show home button
8. Set your home page to brave.com and enable "show home button"
9. Erase out the value in home page and confirm "show home button" is:
  - is turned off
  - disabled completely (not clickable)
10. Type a non-whitespace character or characters into home page box
11. Confirm the "show home button" is clickable again

### overall general settings tab
12. Ensure screen matches corresponding parts of mockup shown in https://github.com/brave/browser-laptop/issues/2108
13. Ensure screen matches corresponding parts of mockup shown in https://github.com/brave/browser-laptop/issues/2111

## Screenshots
cc: @bradleyrichter 

**Before**
![screen shot 2016-10-08 at 2 31 34 pm](https://cloud.githubusercontent.com/assets/4733304/19216081/4d91d4c2-8d64-11e6-91a2-a0c5d02a0296.png)

**After**
![screen shot 2016-10-08 at 2 31 54 pm](https://cloud.githubusercontent.com/assets/4733304/19216078/4813d964-8d64-11e6-9b72-deba9361bf6a.png)
